javac-plugin/build.gradle.kts

```diff
@@ -0,0 +1,31 @@
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(8))
+    }
+}
+
+val java8 = javaToolchains.compilerFor { languageVersion.set(JavaLanguageVersion.of(8)) }.get()
+
+dependencies {
+    compileOnly(rootProject.sourceSets["main"].output)
+    compileOnly(rootProject.sourceSets["shared"].output)
+
+    implementation(files("${java8.metadata.installationPath}/lib/tools.jar"))
+
+    testImplementation("org.junit.jupiter:junit-jupiter:5.10.2")
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+}
+
+tasks.compileTestJava {
+    classpath += rootProject.tasks["shadowJar"].outputs.files
```

Comment:
you should use the normal `jar` task here. and supply the `testApi` task's jar.
see:  https://github.com/unimined/JvmDowngrader/blob/cb724f459f32758e5b1d39a9cb5dc06e2aa2bb58/testing/build.gradle.kts#L73
which you can pass to the api flag.

this will make testing faster and not requiring the coverage report for testing.